src/plugins/intel_gna/src/transformations/matops_to_dwsc.cpp

```diff
@@ -0,0 +1,261 @@
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/cc/ngraph/itt.hpp>
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+// Check if the previous node is convolution, groupconvolution or matmul followed by add
+static bool IsFusable(const Output<Node>& parent) {
+    if (nullptr == std::dynamic_pointer_cast<ngraph::opset1::GroupConvolution>(parent.get_node()->shared_from_this()) ||
+        nullptr == std::dynamic_pointer_cast<ngraph::opset1::Convolution>(parent.get_node()->shared_from_this()) ||
+        nullptr == std::dynamic_pointer_cast<ngraph::opset1::MatMul>(parent.get_node()->shared_from_this()))
+        return false;
+    return true;
+}
+
+static bool Decompose(std::shared_ptr<ov::Node> math_node) {
+    const Output<Node>& input = math_node->input_value(0);
+    const Output<Node>& params = math_node->input_value(1);
+
+    auto input_shape = input.get_shape();
+    auto params_shape = params.get_shape();
+    auto auto_broadcast = math_node->get_autob();
+    auto output_shape = math_node->get_output_shape(0);
+
+    if (input_shape.size() == 0)
+        return false;
+
+    uint64_t N, C, H, W;
+    switch (input_shape.size()) {
+    case 4:
+        N = input_shape[0];
+        C = input_shape[1];
+        H = input_shape[2];
+        W = input_shape[3];
+        break;
+    case 2:
+        N = 1;
+        C = input_shape[1];
+        H = 1;
+        W = input_shape[0];
+        break;
+    default:
+        return false;
+    }
+
+    uint64_t N_params, C_params, H_params, W_params;
+    switch (params_shape.size()) {
+    case 4:
+        N_params = params_shape[0];
+        C_params = params_shape[1];
+        H_params = input_shape[2];
+        W_params = input_shape[3];
+        break;
+    case 2:
+        N_params = params_shape[0];
+        C_params = params_shape[1];
+        H_params = 1;
+        W_params = 1;
+        break;
+    case 1:
+        N_params = 1;
+        C_params = params_shape[0];
+        H_params = 1;
+        W_params = 1;
+        break;
+    default:
+        return false;
+    }
+
+    // getting the parent(previous) node
+    const Output<Node>& parent = math_node->input_value(0);
+    if (nullptr != std::dynamic_pointer_cast<ngraph::opset1::Multiply>(math_node) && IsFusable(parent))
+        return false;
+
+    auto G = C;             // number of groups
+    auto Co = (uint64_t)1;  // no. kernels in a group
+    auto Ci = (uint64_t)1;  // no. input channels in a group
+    auto Kh = (uint64_t)1;  // kernel height
+    auto Kw = (uint64_t)1;  // kernel width
+
+    // Initialize dwsc (group convolution) kernel weights as 1s
+    std::vector<float> dwsc_weights(Kh * Kw * G, 0.0f);
+    float* dwsc_weight_ptr = dwsc_weights.data();
+    if (nullptr == std::dynamic_pointer_cast<ngraph::opset1::Multiply>(math_node)) {
+        for (uint32_t i = 0; i < Kh * Kw * G; i++)
```

Comment:
Perhaps you could consider using something that is easier to read?
std::fill(dwsc_weights.begin(), dwsc_weights.end(), 1.0f);

Reply:
Done